### ISSUE TITLE
Support multiple schemas in `@response`?

### ISSUE THREAD

**Issue by anilravuri** [miguelgrinberg/APIFairy#64]
This is a requirement we had in our team where we have a different set of responses (thus multiple schemas) for a given endpoint. I tried [marshmallow-oneofschema]( https://github.com/marshmallow-code/marshmallow-oneofschema) but that does not seem to work. Any thoughts on this?

**Comment by miguelgrinberg** [comment]
I don't consider the idea of multiple schemas a good API design practice, so I really have no direct experience (or interest) in implementing this. But marshmallow supports nesting schemas, which may give you something close to what you need. Or else you can hand modify the OpenAPI data structure in a `process_apispec` callback and make it be exactly what you need.

**Comment by sgavil** [comment]
@miguelgrinberg What would be the approach if I wanted to return a schema if everything goes well but some kind of failed schema if something goes wrong?

**Comment by miguelgrinberg** [comment]
@sgavil Your question makes me think that you think APIFairy somehow prevents you from returning an error response based on a schema. This is actually incorrect, you can returning anything you want in an error handler, including a serialized Marshmallow schema.

In terms documenting your error schemas, see the [@other_responses](https://apifairy.readthedocs.io/en/latest/decorators.html#other-responses) decorator.

**Comment by sersorrel** [comment]
The thing that trips _me_ up with this is that, say you have a flask-httpauth error handler which might return a 401, 403, whatever, and you apply this flask-httpauth object to a route with `@apifairy.authenticate`: now you must also remember to apply `@apifairy.other_responses` to the _route_ to show what the _error handler_ could return, and you have to do this for every route you use the error handler on. That makes it very easy to forget the `@apifairy.other_responses` and accidentally give the impression that a route will never fail, when really it can return the same authentication failures as any of your other authenticated routes, and makes it very difficult to ever change what your error handler produces, because you would have to update the `@other_response` decorator on every route it is used on.

**Comment by miguelgrinberg** [comment]
@sersorrel You can create your own custom authentication decorator that passes all its arguments through to `apifairy.authenticate` and also includes `apifairy.other_responses` with the set of canned error response you'd like all your authenticated endpoints to have.

**Comment by sgavil** [comment]
@miguelgrinberg But if I add the `@response` with a  schema and then I return a different failure schema, the failure schema gets converted into the schema I've specified in the response decorator. Am I doing something wrong?

**Comment by miguelgrinberg** [comment]
@sgavil Yes, you should return your error responses in an error handler, not in from the route function.
